### PR TITLE
COMMERCE-10762

### PR DESCRIPTION
**Pull request by elocamp**
The issues were being reproduced because there wasn't a adequate check for billing and shippingAllowed in the query search builder.
Proposed fix is to add this check and consider if just one of or both of them are true to return the query search accordingly.

Related tickets: [COMMERCE-10760](https://issues.liferay.com/browse/COMMERCE-10760), [COMMERCE-10762](https://issues.liferay.com/browse/COMMERCE-10762)

### CODE REVIEW

**Comment by elocamp** [comment]
ci:test:sf

**Comment by elocamp** [comment]
ci:test:relevant

**Comment by elocamp** [comment]
ci:test:commerce

**Comment by brianikim** [comment]
@andrea-ale-sbarra , COMMERCE-10760 and COMMERCE-10762 are both solved by this PR.

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/country/CommerceCountryManagerImpl.java [hunk 1]
```diff
@@ -191,10 +191,16 @@ private GroupByStep _getGroupByStep(
 					channelFilterPredicate);
 
 				predicate = predicate.and(groupFilterPredicate);
-				predicate = predicate.and(
-					CountryTable.INSTANCE.billingAllowed.eq(billingAllowed));
-				predicate = predicate.and(
-					CountryTable.INSTANCE.shippingAllowed.eq(shippingAllowed));
+
+				if (billingAllowed) {
+					predicate = predicate.and(
```
Comment (brianikim) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/country/CommerceCountryManagerImpl.java:196]: Change to following please:
```
				predicate = predicate.and(
						CountryTable.INSTANCE.billingAllowed.eq(billingAllowed));
```

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/country/CommerceCountryManagerImpl.java [hunk 2]
```diff
@@ -191,10 +191,16 @@ private GroupByStep _getGroupByStep(
...
+						CountryTable.INSTANCE.billingAllowed.eq(true));
+				}
+
+				if (shippingAllowed) {
+					predicate = predicate.and(
+						CountryTable.INSTANCE.shippingAllowed.eq(true));
```
Comment (brianikim) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/country/CommerceCountryManagerImpl.java:202]: Same here:
```
		predicate = predicate.and(
						CountryTable.INSTANCE.shippingAllowed.eq(shippingAllowed));
```

**Comment by elocamp** [comment]
@brianikim, please take a look at the changes

**Comment by brianikim** [comment]
@andrea-ale-sbarra , this needs your review.

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/country/CommerceCountryManagerImpl.java (hunk 1 above)
Comment (brianikim) [reply modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/country/CommerceCountryManagerImpl.java:196]: @andrea-ale-sbarra , could you double check this for me in terms of SF. Do we pass in the variable boolean or do we just set it to "true". I am not sure which one is the correct one actually in terms of SF.

Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/country/CommerceCountryManagerImpl.java:196]: I think it's fine to use the variable in this case

**Review by andrea-ale-sbarra — APPROVED** [review]

**Comment by brianikim** [comment]
ci:stop

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:stable

**Comment by victoroscuro** [comment]
ci:test:commerce-functional

**Comment by elocamp** [comment]
ci:close

**Pull request closed without merging**